### PR TITLE
Pure Go

### PR DESCRIPTION
**Pull request by xtellurian**
This PR is the main PR for the pure go work into master.

It includes:
 * Changed build definition
 * Go implementation
 * Removal of python
 * Upgrade to KubeBuilder 2

### CODE REVIEW

**Comment by xtellurian** [comment]
Tests in ` microsoft/azure-databricks-operator/pkg/controller/` seem to be failing.

**Comment by xtellurian** [comment]
#24

**Comment by xtellurian** [comment]
Above issue is resolved by adding DATABRICKS_HOST and DATABRICKS_TOKEN to the test environment.

**Comment by xtellurian** [comment]
Currently tests are failing with the following error:

```
--- FAIL: TestReconcile (104.09s)
    notebookjob_controller_test.go:146: 
        Timed out after 60.008s.
        Error: Expected an error, got nil
FAIL
```

**Comment by xinsnake** [comment]
Looking into it now. This issue seems to be intermittent.

**Comment by Azadehkhojandi** [comment]
Please update readme while I'm reviewing it

**Comment by xtellurian** [comment]
Tests are now passing.

We're going to add some utility binaries to the repository using git LFS. This will enable us to manage the binary versions more reliably, and enable us to run kustomize in the build pipeline. We need this to generate the `setup.yaml` people are expecting in the github release

**Comment by xinsnake** [comment]
It seems that LFS support in the pipeline is not enabled. @xtellurian  can you please check the DevOps build?

**Comment by xinsnake** [comment]
@xtellurian  Now upgraded to use kubebuilder v2 and embedded Kubernetes API server. Now build time is reduced from 9+ minutes to 6+ minutes. We should be able to further reduce it as currently, we are testing outside the container but building inside the container, which should be made consistent and will also save time.

```
ok  	github.com/microsoft/azure-databricks-operator/api/v1	7.716s	coverage: 20.4% of statements
ok  	github.com/microsoft/azure-databricks-operator/controllers	46.205s	coverage: 80.9% of statements
```

File: config/default/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: mcr.microsoft.com/k8s/azure-databricks/operator:latest
```
Comment (Azadehkhojandi) [inline config/default/manager_image_patch.yaml]: I need to remember to change release pipeline then

**Comment by Azadehkhojandi** [comment]
Awesome work @xinsnake :)
I'll update release pipeline

**Pull request merged**